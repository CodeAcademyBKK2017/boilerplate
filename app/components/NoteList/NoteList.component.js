import PropTypes from 'prop-types';
import React, {Component} from 'react';
import styles from './NoteList.style';
import {
  FlatList, Text, View
} from 'react-native';

export default class NoteList extends Component {
  renderItem = ({item}) => (
    <View style={styles.itemContainer}>
      <Text style={styles.itemTitle}>{item.title}</Text>
      <Text style={styles.itemContent}>{item.content}</Text>
    </View>
  );

  render () {
    return (
      <View style={styles.container}>
        <Text style={styles.noteTitle}>Notes:</Text>
        <FlatList
          data={this.props.data}
          renderItem={this.renderItem}/>
      </View>
    );
  }
}

NoteList.propTypes = {
  data: PropTypes.array.isRequired
};

NoteList.defaultProps = {
  data: []
};
