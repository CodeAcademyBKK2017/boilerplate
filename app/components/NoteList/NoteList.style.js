import {
  StyleSheet
} from 'react-native';

const styles = StyleSheet.create({
  container: {
    flex: 1,
    width: '100%'
  },
  noteTitle: {
    fontWeight: 'bold'
  },
  itemContainer: {
    paddingVertical: 10
  },
  itemTitle: {
    fontWeight: 'bold',
    fontSize: 14,
    paddingBottom: 10
  },
  itemContent: {
    fontSize: 12
  }
});

export default styles;
