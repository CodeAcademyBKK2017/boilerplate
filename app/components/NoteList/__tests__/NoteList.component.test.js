import NoteList from '../NoteList.component';
import React from 'react';

// Note: test renderer must be required after react-native.
import renderer from 'react-test-renderer';
import {shallow} from 'enzyme';
import 'react-native';

describe('NoteList', () => {
  it('renders correctly', () => {
    const snapshot = renderer.create(<NoteList />).toJSON();
    expect(snapshot).toMatchSnapshot();
  });

  it('renderItem with even', () => {
    const item = {
      title: 'test title',
      content: 'test content',
      isEven: true
    };

    const wrapper = shallow(<NoteList />);
    const instance = wrapper.instance();
    const snapshot = instance.renderItem({item});
    expect(snapshot).toMatchSnapshot();
  });

  it('renderItem with odd', () => {
    const item = {
      title: 'test title',
      content: 'test content',
      isEven: false
    };

    const wrapper = shallow(<NoteList />);
    const instance = wrapper.instance();
    const snapshot = instance.renderItem({item});
    expect(snapshot).toMatchSnapshot();
  });

  it('onOpenOverlay', () => {
    const item = {
      title: 'test title',
      content: 'test content',
      isEven: true
    };

    const wrapper = shallow(<NoteList />);
    const instance = wrapper.instance();
    const curryFunc = instance.onOpenOverlay(item);
    curryFunc();
    
    const expected = {
      modalVisible: true,
      selectedNoteItem: item
    };
    expect(instance.state).toEqual(expected);
  });

  it('onCloseOverlay', () => {
    const wrapper = shallow(<NoteList />);
    const instance = wrapper.instance();
    instance.onCloseOverlay();

    const expected = {
      modalVisible: false,
      selectedNoteItem: {}
    };
    expect(instance.state).toEqual(expected);
  });
});
