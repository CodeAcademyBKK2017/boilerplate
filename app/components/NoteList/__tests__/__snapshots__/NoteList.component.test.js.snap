// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`NoteList renderItem with even 1`] = `
<PlatformTouchable
  background={Object {}}
  onPress={[Function]}
  style={
    Array [
      Object {},
      Object {
        "backgroundColor": "gray",
      },
    ]
  }
>
  <View
    style={
      Object {
        "paddingVertical": 10,
      }
    }
  >
    <Text
      accessible={true}
      allowFontScaling={true}
      ellipsizeMode="tail"
      style={
        Object {
          "fontSize": 14,
          "fontWeight": "bold",
          "paddingBottom": 10,
        }
      }
    >
      test title
    </Text>
    <Text
      accessible={true}
      allowFontScaling={true}
      ellipsizeMode="tail"
      style={
        Object {
          "fontSize": 12,
        }
      }
    >
      test content
    </Text>
  </View>
</PlatformTouchable>
`;

exports[`NoteList renderItem with odd 1`] = `
<PlatformTouchable
  background={Object {}}
  onPress={[Function]}
  style={
    Array [
      Object {},
      Object {
        "backgroundColor": "white",
      },
    ]
  }
>
  <View
    style={
      Object {
        "paddingVertical": 10,
      }
    }
  >
    <Text
      accessible={true}
      allowFontScaling={true}
      ellipsizeMode="tail"
      style={
        Object {
          "fontSize": 14,
          "fontWeight": "bold",
          "paddingBottom": 10,
        }
      }
    >
      test title
    </Text>
    <Text
      accessible={true}
      allowFontScaling={true}
      ellipsizeMode="tail"
      style={
        Object {
          "fontSize": 12,
        }
      }
    >
      test content
    </Text>
  </View>
</PlatformTouchable>
`;

exports[`NoteList renders correctly 1`] = `
<View
  style={
    Object {
      "flex": 1,
      "width": "100%",
    }
  }
>
  <Text
    accessible={true}
    allowFontScaling={true}
    ellipsizeMode="tail"
    style={
      Object {
        "fontWeight": "bold",
      }
    }
  >
    Notes:
  </Text>
  <RCTScrollView
    data={Array []}
    disableVirtualization={false}
    getItem={[Function]}
    getItemCount={[Function]}
    horizontal={false}
    initialNumToRender={10}
    keyExtractor={[Function]}
    maxToRenderPerBatch={10}
    numColumns={1}
    onContentSizeChange={[Function]}
    onEndReachedThreshold={2}
    onLayout={[Function]}
    onMomentumScrollEnd={[Function]}
    onScroll={[Function]}
    onScrollBeginDrag={[Function]}
    onScrollEndDrag={[Function]}
    renderItem={[Function]}
    scrollEventThrottle={50}
    stickyHeaderIndices={Array []}
    updateCellsBatchingPeriod={50}
    viewabilityConfigCallbackPairs={Array []}
    windowSize={21}
  >
    <View />
  </RCTScrollView>
  <Modal
    animationOutType="fadeOut"
    animationType="none"
    hardwareAccelerated={false}
    onClose={[Function]}
    onRequestClose={[Function]}
    transparent={true}
    visible={false}
  >
    <View
      accessibilityComponentType={undefined}
      accessibilityLabel={undefined}
      accessibilityTraits={undefined}
      accessible={true}
      collapsable={undefined}
      hitSlop={undefined}
      nativeID={undefined}
      onLayout={undefined}
      onResponderGrant={[Function]}
      onResponderMove={[Function]}
      onResponderRelease={[Function]}
      onResponderTerminate={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
      style={
        Object {
          "backgroundColor": "rgba(37, 8, 10, 0.78)",
          "flex": 1,
          "justifyContent": "center",
          "opacity": 0,
          "padding": 20,
        }
      }
      testID={undefined}
    >
      <View
        accessibilityComponentType={undefined}
        accessibilityLabel={undefined}
        accessibilityTraits={undefined}
        accessible={true}
        hitSlop={undefined}
        nativeID={undefined}
        onLayout={undefined}
        onResponderGrant={[Function]}
        onResponderMove={[Function]}
        onResponderRelease={[Function]}
        onResponderTerminate={[Function]}
        onResponderTerminationRequest={[Function]}
        onStartShouldSetResponder={[Function]}
        style={
          Array [
            Object {
              "alignItems": "center",
              "backgroundColor": "white",
              "padding": 20,
            },
            undefined,
          ]
        }
        testID={undefined}
      >
        <Text
          accessible={true}
          allowFontScaling={true}
          ellipsizeMode="tail"
        />
        <Text
          accessible={true}
          allowFontScaling={true}
          ellipsizeMode="tail"
        />
      </View>
    </View>
  </Modal>
</View>
`;
